src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,91 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+import org.reflections.Reflections;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplV2(interfaceClazz);
+        isImplClassWithAnotation(clazz);
+        Object clazzImplInstance = null;
+        Field[] clazzDeclaratedFields = clazz.getDeclaredFields();
+        for (Field clazzField : clazzDeclaratedFields) {
+            if (clazzField.isAnnotationPresent(Inject.class)) {
+                Object clazzFieldInstance = getInstance(clazzField.getType());
+                clazzImplInstance = createNewInstance(clazz);
+                clazzField.setAccessible(true);
```

Comment:
put this line inside the try block, cause `setAccessible()` would throw an exception too. [See documentation](https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/AccessibleObject.html#setAccessible(boolean))

Reply:
++